pkg/kn/flags/podspec_helper.go

```diff
@@ -865,3 +889,134 @@ func decodeContainersFromFile(filename string) (*corev1.PodSpec, error) {
 	}
 	return podSpec, nil
 }
+
+// =======================================================================================
+// Probes
+
+// resolveProbe parses probes as a string
+// It's split into two functions:
+//   - resolveProbeOptions() -> common probe opts
+//   - resolveProbeHandler() -> probe handler [HTTPGet, Exec, TCPSocket]
+// Format:
+//	- [http,https]:host:port:path;<common_opts>
+//	- exec:cmd,cmd,...;<common_opts>
+//  - tcp:host:port;<common_opts>
+// Common opts (comma separated, case insensitive):
+//	- <probe_opts>;InitialDelaySeconds=<int_value>,FailureThreshold=<int_value>,
+//  	SuccessThreshold=<int_value>,PeriodSeconds==<int_value>,TimeoutSeconds=<int_value>
+func resolveProbe(probeString string) (*corev1.Probe, error) {
+	parts := strings.Split(probeString, ";")
```

Comment:
Do you think that a value can contain also a ; ? Like as part of an argument for an exec probe ? I think this is quite unlikely, but not sure whether we should allows for escaping characters here. Maybe just continue like here and wait until the first bug reports occurs :-)

Reply:
There's a bit of trimming done later in the other two `resolve*` functions. That space wouldn't make it too far. :)

Reply:
Ok, let's keep it for now like it is.